### ISSUE TITLE
StoryBlock - Embedding a content item to itself causes stackoverflow

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#23846]
### Proposed Objective

Accessibility (WCAG)

### Proposed Priority

Priority 2 - Important

### dotCMS Version

23.01

### Problem Statement

You can add a content item to itself in a storyblock field.  The issue is that it breaks the content.

```
at com.dotcms.content.elasticsearch.business.ESContentFactoryImpl.find(ESContentFactoryImpl.java:860) ~[classes/:?]
at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.find(ESContentletAPIImpl.java:369) ~[classes/:?]
at com.dotmarketing.portlets.contentlet.business.ContentletAPIInterceptor.find(ContentletAPIInterceptor.java:698) ~[classes/:?]
at com.dotcms.contenttype.business.StoryBlockAPIImpl.refreshBlockEditorDataMap(StoryBlockAPIImpl.java:268) ~[classes/:?]
at com.dotcms.contenttype.business.StoryBlockAPIImpl.refreshStoryBlockMap(StoryBlockAPIImpl.java:117) ~[classes/:?]
at com.dotcms.contenttype.business.StoryBlockAPIImpl.refreshStoryBlockValueReferences(StoryBlockAPIImpl.java:71) ~[classes/:?]
at com.dotcms.contenttype.business.StoryBlockAPIImpl.lambda$0(StoryBlockAPIImpl.java:48) ~[classes/:?]
```

### Steps to Reproduce

1. In starter, create a new blog entry
2. Save it
3. In the blog content (the main storyblock field) add the blog content to itself.
4. Save it (fails) and close
5. try to reopen it.

### Acceptance Criteria

Content should not be able to be added to itself.

**Comment by wezell** [comment]
Bumping to important as it is unrecoverable once this happens.  The content can't be opened, edited or deleted.

**Comment by binarybasher** [comment]
This is a real issue, even if it's not linked too itself, it can be linked to another content-let which links back.

Editor's should not be able to cause this unrecoverable failure, it affects the indexing and other stuff.

**Comment by swicken-dotcms** [comment]
Related ticket: https://dotcms.zendesk.com/agent/tickets/111131

**Comment by manuelrojas** [comment]
**Passed Internal QA** Docker Image: [dotcms/dotcms:master_1fc8dba8_SNAPSHOT]



https://github.com/dotCMS/core/assets/3438705/d1deb9df-a99b-4490-a012-c334acf8afdb

**Comment by josemejias11** [comment]
**Approved QA** - Tested on **23.06_8c0a542e_SNAPSHOT** // Docker // macOS 13.0 // FF v113.0

**Comment by erickgonzalez** [comment]
Excluded of 22.03 LTS
